examples/gocean/eg5/nan/README.md

```diff
@@ -23,24 +23,20 @@ This script uses PSyclone's ``NanTestTrans`` to instrument the two
 invokes in the ``test.x90`` source file.
 
 The source code computes divisions by 0 on the diagonals, resulting in
-a invalid numbers (Infinity).
+invalid numbers (Infinity).
 
 ## Running
 ```
 $ ./nan_test
 ...
 Allocating C-T field with bounds: (1:   6,1:   6), internal region is (2:   4,2:   4)
- PSyData: Variable a_fld has the invalid value
-                  Infinity  at index/indices            1           1
- mainupdate
- PSyData: Variable a_fld has the invalid value
-                  Infinity  at index/indices            2           2
- mainupdate
+PSyData: Variable a_fld has the invalid value Inf at index/indices 1 1 in module 'main' region 'update'.
+PSyData: Variable a_fld has the invalid value Inf at index/indices 2 2 in module 'main' region 'update'.
 
 ```
-After calling the kernel ``mainupdate``, five warnings are being printed,
-indicating that the field ``a_fld`` has infinity as value on the diagonals
-(i.e. indices 1 1, ..., 5 5)
+After calling the kernel ``mainupdate``, five warnings are printed,
```

Comment:
Unfortunately I can't reproduce this! I can build and run the example but no warnings are produced. If I examine the generated PSy code then I see no variables are actually provided for checking:

      TYPE(nan_test_PSyDataType), save, target :: nan_test_psy_data

      CALL nan_test_psy_data % PreStart("main", "init", 0, 0)
      DO j = a_fld%whole%ystart, a_fld%whole%ystop, 1
        DO i = a_fld%whole%xstart, a_fld%whole%xstop, 1
          CALL init_field_code(i, j, a_fld%data, 1.0)
        END DO
      END DO
      DO j = b_fld%whole%ystart, b_fld%whole%ystop, 1
        DO i = b_fld%whole%xstart, b_fld%whole%xstop, 1
          CALL init_field_code(i, j, b_fld%data, 2.0)
        END DO
      END DO
      CALL nan_test_psy_data % PostEnd

Please could you investigate?

Reply:
The NanNode did not have a `lower_to_language_level` function. As such it worked when using the old `gen` method, but not when using the psyir FortranWriter (which is done under the hood in the example).

I've added that missing function, created a new test (very similar to an existing test, but using the FortranWriter for output), and verified that the test would have triggered (without the missing function), and works now.

There is some concern that other codes might have the same issue (extract has the lower function, read_only has, profile doesn't need it (it falls back to the base function, which works since profile does not use any variables)). I didn't check any further (e.g. if they have tests that uses a visitor - I doubt that somehow :) ). But I assume that soonish we will disable the old-style gen approach for gocean, and then we can update all these tests just once (instead of now duplicating them to use old style and visitor style, then later removing them). Let me know if you want a ticket to keep track of that.

Reply:
It all works for me now thanks. As discussed on Teams, once we've done away with gocean0.1 and Sergi has finished refactoring the OpenCL support then we can remove the `gen` functionality for GOcean.